dev-tools/mage/fields.go

```diff
@@ -143,13 +143,33 @@ func GenerateModuleFieldsGo(moduleDir string) error {
 // GenerateModuleIncludeListGo generates an include/list.go file containing
 // a import statement for each module and dataset.
 func GenerateModuleIncludeListGo() error {
-	return GenerateIncludeListGo(nil, []string{"module"})
+	return GenerateIncludeListGo(nil, []string{"module"}, nil, "include/list.go", "")
+}
+
+// GenerateOSSMetricbeatModuleIncludeListGo generates include/list_{suffix}.go files containing
+// a import statement for each module and dataset.
+func GenerateOSSMetricbeatModuleIncludeListGo() error {
+	err := GenerateIncludeListGo(
+		nil, []string{"module"},
+		[]string{"module/docker", "module/kubernetes"},
+		"include/list_common.go", "")
+	if err != nil {
+		return err
+	}
+	err = GenerateIncludeListGo(
+		nil, []string{"module/docker", "module/kubernetes"},
+		nil,
+		"include/list_docker.go", "\n// +build linux darwin windows\n")
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
 // GenerateIncludeListGo generates an include/list.go file containing imports
 // for the packages that match the paths (or globs) in importDirs (optional)
 // and moduleDirs (optional).
-func GenerateIncludeListGo(importDirs []string, moduleDirs []string) error {
+func GenerateIncludeListGo(importDirs []string, moduleDirs []string, modulesToExclude []string, outfile string, buildTags string) error {
```

Comment:
This is starting to have quite a lot of different options, we could consider to have an options struct to pass as single parameter here. this would also allow us to set defaults, like the `include/list.go` that most beats use.